sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyLessECSignature.java

```diff
@@ -0,0 +1,168 @@
+        }
+        this.keyVaultDigestName = keyVaultDigestName;
+    }
+
+    //add this for test
+    void setKeyVaultClient(KeyVaultClient keyVaultClient) {
+        this.keyVaultClient = keyVaultClient;
+    }
+
+    @Override
+    protected void engineInitSign(PrivateKey privateKey) {
+        engineInitSign(privateKey, null);
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as RSAPSSSignature#engineInitSign.
+    @Override
+    protected void engineInitSign(PrivateKey privateKey, SecureRandom random) {
+        if (privateKey instanceof KeyVaultPrivateKey) {
+            keyId = ((KeyVaultPrivateKey) privateKey).getKid();
+        } else {
+            throw new UnsupportedOperationException("engineInitSign() not supported which private key is not instance of KeyVaultPrivateKey");
+        }
+    }
+
+    /**
+     * Get the message digest value.
+     * @return the message digest value.
+     */
+    protected byte[] getDigestValue() {
+        return messageDigest.digest();
+    }
+
+    @Override
+    protected void engineUpdate(byte b) {
+        messageDigest.update(b);
+    }
+
+    @Override
+    protected void engineUpdate(byte[] b, int off, int len) {
+        messageDigest.update(b, off, len);
+    }
+
+    @Override
+    protected void engineUpdate(ByteBuffer byteBuffer) {
+        int len = byteBuffer.remaining();
+        if (len <= 0) {
+            return;
+        }
+        messageDigest.update(byteBuffer);
+    }
+
+    @Override
+    protected byte[] engineSign() {
+        byte[] mHash = getDigestValue();
+        String encode = Base64.getEncoder().encodeToString(mHash);
+        byte[] encrypted = keyVaultClient.getSignedWithPrivateKey(keyVaultDigestName, encode, keyId);
+        return encodeByte(encrypted);
+    }
+
+    // After throw UnsupportedOperationException, other methods will be called.
+    // such as ECDSASignature#engineSetParameter.
+    @Override
+    protected void engineSetParameter(AlgorithmParameterSpec params)
+        throws InvalidAlgorithmParameterException {
+        if (params != null && !(params instanceof ECParameterSpec)) {
+            throw new InvalidAlgorithmParameterException("No parameter accepted");
+        }
+        signatureParameters = (ECParameterSpec) params;
+    }
+
+    @Override
+    protected AlgorithmParameters engineGetParameters() {
+        if (signatureParameters == null) {
+            return null;
+        }
+        try {
+            AlgorithmParameters parameters = AlgorithmParameters.getInstance("EC");
+            parameters.init(signatureParameters);
+            return parameters;
+        } catch (Exception e) {
+            // should never happen
+            throw new ProviderException("Error retrieving EC parameters", e);
+        }
+    }
+
+    /**
+     * key vault SHA384
+     */
+    public static final class KeyVaultSHA384 extends KeyVaultKeyLessECSignature {
```

Comment:
How about:
1. Move these 3 classes out of `KeyVaultKeyLessECSignature`.
2. Rename `KeyVaultSHA384` to `KeyVaultKeyLessEcSha384Signature`.